glean-core/src/database/mod.rs

```diff
@@ -78,6 +84,40 @@ impl Database {
         }
     }
 
+    /// Loads Lifetime::Ping data from rkv to memory,
+    /// if `delay_ping_lifetime_io` is set to true.
+    ///
+    /// Does nothing if it isn't or if there is not data to load.
+    fn load_ping_lifetime_data(&self) {
+        if let Some(ping_lifetime_data) = &self.ping_lifetime_data {
+            let mut data = ping_lifetime_data
+                .write()
+                .expect("Can't read ping lifetime data");
+
+            let store: SingleStore = unwrap_or!(
+                self.rkv
+                    .open_single(Lifetime::Ping.as_str(), StoreOptions::create()),
+                return
+            );
+
+            let reader = unwrap_or!(self.rkv.read(), return);
+            let mut iter = unwrap_or!(store.iter_start(&reader), return);
+
+            while let Some(Ok((metric_name, value))) = iter.next() {
+                let metric_name = match str::from_utf8(metric_name) {
+                    Ok(metric_name) => metric_name.to_string(),
+                    _ => continue,
+                };
+                let metric: Metric = match value.expect("Value missing in iteration") {
+                    rkv::Value::Blob(blob) => unwrap_or!(bincode::deserialize(blob), continue),
+                    _ => continue,
+                };
+
+                data.insert(metric_name, metric.clone());
```

Comment:
Do we need to clone here? I think `metric` is already an owned value, so we can avoid the clone.